### PR TITLE
Add basic test coverage

### PR DESCRIPTION
**Pull request by merlinran**
This PR:
1. Sets GitHub workflow to run tests for new commits and pull requests. Also added code coverage.
2. Adds two basic tests for main process and remote control process.
3. Allows running tests locally with code coverage printed out.

### CODE REVIEW

File: vehicle/start_main.sh [hunk 1]
```diff
@@ -1,5 +1,4 @@
-sleep 5
 source /home/pi/.bashrc
 PATH=/home/pi/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/games:/usr/games
 tmux  pipe-pane -o 'cat >>/home/pi/logs/tmux_output_$(date +"%Y_%m_%d_%I_%M_%p").#S:#I-#P'
-python3 /home/pi/vehicle/master_process.py
+python3 /home/pi/vehicle/master_process.py $@
```
Comment (merlinran) [inline vehicle/start_main.sh:4]: `$@` passes the parameters to the process (if any), so `start_main_sim.sh` is no longer required.

File: vehicle/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import pickle
+import pytest
+
+
+@pytest.fixture(scope="session")
```
Comment (merlinran) [inline vehicle/tests/conftest.py:5]: These fixtures are used by the tests. The magic strings are extracted from Redis.

File: vehicle/tests/test_main_process.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    context = zmq.Context()
+    socket = context.socket(zmq.REP)
+    if port is None:
+        port = socket.bind_to_random_port("tcp://127.0.0.1")
+    else:
+        socket.bind("tcp://127.0.0.1:{}".format(port))
+    return socket, port
+
+
+def test_main(fixture_path, fixture_robot_command):
```
Comment (merlinran) [inline vehicle/tests/test_main_process.py:24]: This very preliminary test makes sure the main process can be run and stops gracefully. More can be add later but this is a start point. Note that `pytest` does the magic of making the fixtures available here.

File: vehicle/voltage_monitor_process.py [hunk 4]
```diff
@@ -50,67 +50,67 @@ class VoltageSampler():
...
-                    cell1, cell2, cell3, total))
-                print(' Input Voltage Values: | {0:>6} | {1:>6} | {2:>6} | {3:>6} |'.format(
-                    *input_voltages))
-                #print(' Raw Values: | {0:>6} | {1:>6} | {2:>6} | {3:>6} |'.format(*values))
-            # Pause for half a second.
-            time.sleep(1)
-
-
-def sampler_loop(master_conn, simulated_hardware=False):
+        else:
```
Comment (merlinran) [inline vehicle/voltage_monitor_process.py:66]: The code diff is messy, but the only change is actually this `else` branch. https://github.com/Twisted-Fields/acorn-precision-farming-rover/pull/12/files?w=1 would make it more clear.

**Review by merlinran — COMMENTED** [review]
@tlalexander I figure it's better to be iterative and add more tests when making changes. 

Also plan to introduce Makefile to replace the various shell scripts if you don't object.
```
make test # replaces run_docker_test.sh
make simulation # run_docker_simulation.sh
make run-vehicle # run_docker_vehicle.sh
make attach-vehicle # attach_docker_vehicle.sh
make run-server # attach_docker_server.sh
make docker-image
...
```

**Comment by tlalexander** [comment]
looks good to me, thank you!!

**Pull request merged**